extensions/jib/deployment/src/main/java/io/quarkus/jib/deployment/JibProcessor.java

```diff
@@ -0,0 +1,108 @@
+
+public class JibProcessor {
+
+    private static final Logger log = Logger.getLogger(JibProcessor.class);
+
+    private static final String MAIN_CLASS = "io.quarkus.runner.GeneratedMain";
+    private static final IsClassPredicate IS_CLASS_PREDICATE = new IsClassPredicate();
+
+    @BuildStep(onlyIf = IsNormal.class) // TODO: are we sure we only want it for normal execution?
+    public void build(
+            // TODO: is this proper build item to consume?
+            JarBuildItem sourceJarBuildItem,
+            OutputTargetBuildItem outputTargetBuildItem, ApplicationInfoBuildItem applicationInfo, JibConfig jibConfig,
+            // TODO figure out the proper output
+            BuildProducer<ArtifactResultBuildItem> producer) {
+        if (!jibConfig.enabled) {
+            log.debug("Jib container image build was disable");
+            return;
+        }
+
+        JibContainerBuilder jibContainerBuilder = createContainerBuilder(sourceJarBuildItem, outputTargetBuildItem);
+        Containerizer containerizer = createContainerizer(jibConfig, applicationInfo);
+        try {
+            JibContainer container = jibContainerBuilder.containerize(containerizer);
+            log.infof("Created container image %s (%s)\n", container.getTargetImage(), container.getDigest());
+        } catch (Exception e) {
+            log.error("Unable to create container image", e);
+        }
+
+        producer.produce(new ArtifactResultBuildItem(null, "containerImage", Collections.emptyMap()));
+    }
+
+    private Containerizer createContainerizer(JibConfig jibConfig, ApplicationInfoBuildItem applicationInfo) {
+        Containerizer containerizer = Containerizer.to(DockerDaemonImage.named(getImageReference(jibConfig, applicationInfo)));
+        containerizer.setToolName("Quarkus");
+        containerizer.addEventHandler(log::debug);
+        return containerizer;
+    }
+
+    private ImageReference getImageReference(JibConfig jibConfig, ApplicationInfoBuildItem applicationInfo) {
+        return ImageReference.of(jibConfig.registry, jibConfig.group + "/" + jibConfig.name.orElse(applicationInfo.getName()),
+                jibConfig.tag.orElse(applicationInfo.getVersion()));
+    }
+
+    private JibContainerBuilder createContainerBuilder(JarBuildItem sourceJarBuildItem,
+            OutputTargetBuildItem outputTargetBuildItem) {
+        try {
+            // not ideal since this has been previously zipped - we would like to just reuse it
+            Path classesDir = outputTargetBuildItem.getOutputDirectory().resolve("jib");
+            ZipUtils.unzip(sourceJarBuildItem.getPath(), classesDir);
+            JavaContainerBuilder javaContainerBuilder = JavaContainerBuilder
+                    .fromDistroless()
+                    .addResources(classesDir, IS_CLASS_PREDICATE.negate())
+                    .addClasses(classesDir, IS_CLASS_PREDICATE)
+                    .setMainClass(MAIN_CLASS);
```

Comment:
As currently Quarkus dont expose a Main class, how the Docker ENTRYPOINT JIB will allow to launch then the Quarkus application using the command `java -cp /app/resources:/app/classes:/app/libs/*` ? @geoand

Reply:
It's all handled automatically because we pass the resources, classes and main class